docs/src/main/sphinx/installation/deployment.rst

```diff
@@ -215,20 +213,14 @@ These properties require some explanation:
   The maximum amount of user and system memory, that a query may use on any one machine,
   where system memory is the memory used during execution by readers, writers, and network buffers, etc.
 
-* ``discovery-server.enabled``:
```

Comment:
it could be valuable to keep the option documented

Reply:
I'd actually like to get rid of this config entirely. We haven't advertised or documented how to use a non-embedded discovery server for years, and I don't see any use case for that. I see discovery as somewhat of an implementation detail that we could simplify in the future -- call it the "coordinator URI" or perhaps just the address and tie into the internal communication config. We could also have tighter integration in the coordinator if it was non-generic.

Reply:
Maybe we decide to remove, or maybe not. Please keep the doc as the property is not being removed.

Reply:
I made it deprecated and removed the usage in `TestingTrinoServer`.

If you feel strongly, I can also add a `@AssertTrue` validation to the config so that it cannot be set to false. This would have a similar effect to making it defunct, but wouldn't break existing configurations.

Reply:
> I can also add a @AssertTrue validation to the config so that it cannot be set to false.

this part i don't understand.

I also do not understand why remove documentation for `discovery-server.enabled` config property, if this config property is not removed. Is it because you don't want to keep documentation for a config that is deprecated?

Reply:
My assertion is that there is no use case for setting `discovery-server.enabled=false`. Do you disagree?

I'm keeping the config option so that we don't break existing configs, but there is no reason that users need to know about it or change the default. If we didn't care about breaking every existing installation, we would simply remove the config.

Reply:
> My assertion is that there is no use case for setting `discovery-server.enabled=false`. Do you disagree?

i neither disagree, nor agree. How can i know, actually?

> If we didn't care about breaking every existing installation

i agree that breaking every existing installation would be terrible.